go/libraries/doltcore/doltdb/doltdb.go

```diff
@@ -1165,6 +1165,12 @@ func (ddb *DoltDB) GC(ctx context.Context, uncommittedVals ...hash.Hash) error {
 	if err != nil {
 		return err
 	}
+
+	uncommittedVals, err := ddb.getGCKeepers(ctx, datasets)
```

Comment:
I think this is unnecessary now, because the roots of each dataset are reachable from the root of datasets map. it doesn't hurt to add them as keeps tho